### ISSUE TITLE
VirtualMachineError: Out of space in CodeCache for adapters in gradle

### ISSUE THREAD

**Issue by dweiss** [apache/lucene#14090]
### Description

I've seen this a few times in jenkins and now github [1]:
[smoke-tester-logs-jdk-21.zip](https://github.com/user-attachments/files/18280106/smoke-tester-logs-jdk-21.zip)

Perhaps we should increase the reserved code cache size? There is a lot going on in the build (groovy); maybe it's really not fitting the defaults for some reason.

https://github.com/apache/lucene/actions/runs/12554562223

**Comment by dweiss** [comment]
I've increased the default to 384m, let's see what happens on CIs.

**Comment by rmuir** [comment]
@dweiss thanks, I've been setting this one explicitly for java stuff at builds at work. 

Problem is that this setting has an ergonomic default, based on how much cpus/ram it sees on build agent (and maybe influenced by cgroup limits if running in container, not sure). It seems to be too low for large java codebases.

So I've just gotten into the habit to set it explicitly to a value needed for things to work correctly all the time and no more troubles.

**Comment by dweiss** [comment]
Yep, good observation. For me, it's not the forced cpu count but the tiered-stop-at-level that causes this to decrease:
```
> java -XX:+PrintFlagsFinal -Xmx1g -XX:TieredStopAtLevel=1 -XX:+UseParallelGC -XX:ActiveProcessorCount=1 --version 2>&1 | grep "ReservedCodeCacheSize"
    uintx ReservedCodeCacheSize                    = 50331648                               {pd product} {default}
```
```
> java -XX:+PrintFlagsFinal  -XX:TieredStopAtLevel=1 --version 2>&1 | grep "ReservedCodeCacheSize"
    uintx ReservedCodeCacheSize                    = 50331648                               {pd product} {default}
> java -XX:+PrintFlagsFinal   --version 2>&1 | grep "ReservedCodeCacheSize"
    uintx ReservedCodeCacheSize                    = 251658240                              {pd product} {ergonomic}
```
 
I was fairly liberal with the setting... maybe I should correct it to, say ~256mb instead of the 384?

**Comment by dweiss** [comment]
I've lowered it to 256 MB. I think it's going to be fine. Thanks for pointing out the ergonomics affect this setting, Rob.

**Comment by dweiss** [comment]
Applied to 10x and main. Closing this issue. We can tweak if we observe more anomalies.
